### ISSUE TITLE
Cannot reproduce the results.

### ISSUE THREAD

**Issue by yfzhang114** [oezyurty/CLUDA#4]
I follow the instructions in the readme document, and use 

ID_src=23
ID_trg=13
python train.py --algo_name cluda --queue_size 8192 --momentum 0.99 --batch_size 32 --eval_batch_size 1024 --num_epochs 10000 --num_steps 10000  --weight_loss_disc 1  --weight_loss_src 0.05 --weight_loss_trg 0.05 --weight_loss_ts 0.1 --learning_rate 1e-4 --dropout 0.2 --weight_decay 1e-4 --num_channels_TCN 64-64-64 --dilation_factor_TCN 3 --kernel_size_TCN 7 --hidden_dim_MLP 128 --path_src ./processed_datasets/HAR --id_src $ID_src --path_trg ./processed_datasets/HAR --id_trg $ID_trg --experiments_main_folder experiment_HAR_$ID_src-$ID_trg --experiment_folder default

however, the result is much lower than the reported results in the paper (76.2), which is 
```
Accuracy score is : 0.6600 
Macro F1 score is : 0.6210 
Weighted F1 score is : 0.6486 
```

**Comment by GuoHaoren** [comment]
python -u main/train.py --algo_name cluda --queue_size 98304 --momentum 0.99 \
--batch_size 128 --eval_batch_size 1 --num_epochs 5000 --num_steps 5000  \
--weight_loss_disc 1  --weight_loss_src 0.05 --weight_loss_trg 0.05 --weight_loss_ts 0.1 \
--learning_rate 1e-4 --dropout 0.2 --weight_decay 1e-4 --num_channels_TCN 64-64-64 \
--dilation_factor_TCN 3 --kernel_size_TCN 7 --hidden_dim_MLP 128 \
--path_src "processed_datasets/HAR/" --id_src 19 --path_trg "processed_datasets/HAR/" --id_trg 25 --experiments_main_folder experiment_HAR_19_25 --experiment_folder default
 
I used the above settings on src 19 and tar 25 (followed by the paper parameter settings)

However the results is much lower than the reported result in the paper (0.932)
VALIDATION TARGET PREDICTIONS
Accuracy score is : 0.7200 
Macro F1 score is : 0.7325 
Weighted F1 score is : 0.7230

**Comment by oezyurty** [comment]
Hi, thank you so much for reaching out to us! 

I just checked my experiment logs on src 19 and tar 25 for HAR, I have identified 3 different parameter settings than what I used for this dataset. I think one of them (queue size) could be the main cause. 

- Queue size: I used 8192, whereas yours is 98304. It's been a long time since I worked on this project, but as far as I remember, the HAR dataset size was not so large. In other words, with the queue size 98304, the CLUDA model might retrieve the embeddings from much earlier epochs (i.e. much earlier versions of the embeddings of the same data points) which can negatively impact the performance. 
- Batch size: For this experiment, I used the batch size 32. I am not sure if the model was that sensitive to the batch size, however, due to the small size of the dataset it may make sense to use smaller batch sizes. 
- Hidden dim of MLP: Mine was 256. I don't think this should affect the performance, but you can try it in any case. 

Last but not the least: Although we have provided train-val-test split processing file for HAR and even fixed the random seed, the final split might be different on different devices. Therefore, your target test split can be slightly different than ours. As a result, you can get slightly different F1 scores, but of course it shouldn't be as different as 0.732 vs. 0.932. 

Also I would like to remind that the reported results are the average of 5 random seeds, i.e. there is no single experiment/configuration that will give the exact same reported result. 

For your convenience, I have provided the entire log for the training and evaluation for `src 19 and tar 25` setting, for a single seed 1111. There you can see that the results are similar to what we have provided. I hope these logs would be helpful to identify the potential differences between our setups so that you can get the similar test results like ours. 

Best regards, 

[commandline_args.txt](https://github.com/oezyurty/CLUDA/files/14520858/commandline_args.txt)
[eval_train.log](https://github.com/oezyurty/CLUDA/files/14520859/eval_train.log)
[train.log](https://github.com/oezyurty/CLUDA/files/14520860/train.log)

**Comment by GuoHaoren** [comment]
Hi, 

Thank you so much for your kind reply! As I followed with the Hyperparameter tuning table in the paper to do the reproduction which is slightly different from the GitHub code, the performance is different. I have done a quick implementation with your suggestion and  I have gotten the relatively same results as you shown in the paper.

The code and paper are both instructive and helpful. 

Best regards.

**Comment by oezyurty** [comment]
I am super glad that it worked out! Best of luck with your project!
